python/tchannel/tornado/connection.py

```diff
@@ -554,4 +574,19 @@ def send_request(self, request):
         future = tornado.gen.Future()
         self._outstanding[request.id] = future
         self.post_request(request)
+
+        def adapt_tracing(f):
+            # fetch the request tracing for response
+            f.result().tracing = request.tracing
```

Comment:
The difference is that you cannot guarantee ordering. If I have two consumers of the future, there's no guarantee which one will be called first. If one of them mutates the future, the second one may or may not get the mutated objects.

You don't modify objects out of band like that. You're opening yourself up to all kinds of race conditions.